household/static/household/js/household.js

```diff
@@ -3,19 +3,28 @@ function amount_convert(value) {
 }
 function overall_chart(container, chartData) {
   var data = [];
+  let xaxis = [];
   for (const [income, value] of Object.entries(chartData)) {
     var region = {
-	    name: income,
-	    type: 'bar',
-	    y: value.y,
-	    x: value.x,
-	    hoverinfo: 'text',
-	    text: value.y.map(amount_convert),
+      name: income,
+      type: 'bar',
+      y: value.y,
+      x: value.x,
+      hoverinfo: 'text',
+      text: value.y.map(amount_convert),
     };
+    xaxis = _.union(xaxis, value.x);
```

Comment:
@michaelglenister should this be inside the for loop and without a push to the xaxis array? I'm not familiar with the `value.x` which may be an array itself?